### DISCUSSION TITLE
请问下，service中调用 ltask.quit() 后，如何重启

### DISCUSSION THREAD

**Discussion by constly** [ejoy/ant#115] [General]
我想把我的逻辑与显示分离，同时也为了方便支持联机，于是我单独起了一个service来跑逻辑层，如下所示
```
local serviceId = ltask.uniqueservice("mini.richman.go|server", ltask.self())
```
我在退出场景时关闭了这个service，代码如下:
```
function S.shutdown()
    quit = {}
    ltask.wait(quit)
    ltask.quit()
end
```
有两个问题:
1. 我调用ltask.quit后，整个service的资源会释放吗
2. 再次进入这个场景后，怎么重启这个服务呢

**Comment by cloudwu** [comment]
service 退出后就真的释放的干干净净了。但是，你不应该用 uniqueservice 启动可以中途退出的服务。通常它约定这个服务是不会退出的，否则会让你的业务很难写对（因为使用这个服务的一方要考虑服务已退出的问题）。

一个可退出的服务应该使用 `ltask.spawn` 来启动。这样启动的服务没有名字，需要自己用 handle 管理。

引擎中通常使用内存最多的是美术资源，而资源是单独服务管理的，并按 LRU 自动管理。所以，退出服务一般并不会帮你节省多少内存，退出服务往往是为了用直接干净的方法清理内存，减少管理复杂度。

如果你需要重启，我建议直接保留，以后继续用就好了。

**Reply by constly** [reply]
明白了，非常感谢！
